09_Hyper-parameter_tuning_using_Ray/index.md

```diff
@@ -1,13 +1,24 @@
 # 09 Hyper-parameter tuning using Ray
 
-## Examples
+## Hands-on exercises
 
-[comment]: <> (List your examples from the lecture here and provide the necessary links to scripts, notebooks, etc. to run them on LUMI)
+In this exercise we perform Hyper-parameter tuning for the model used in lecture 03. The python code in this lecture, `GPT-neo-ray-tune.py` is based on a trimmed down version of `GPT-neo-IMDB-finetuning.py` from lecture 03.
 
-* ...
+1. In order to run the ray example we need to extend the `pytorch_transformers.sif` container with the pip package `ray[tune]`.
 
-## Hands-on exercises
+    * Extend the container via a virtual environment and install `ray[tune]`. You can follow the instructions in [07_Virtual_environments_to_iterate_and_test](https://github.com/Lumi-supercomputer/Getting_Started_with_AI_workshop/blob/main/07_Virtual_environments_to_iterate_and_test/index.md)
+    Note, that you need to source the venv in the batch script.
+
+2. Set up job script for using one whole node for ray tune.
+
+    * Specify slurm parameters in `run.sh` file to use whole node with all available GPUs.
+   
+    * At the end of the file, set up running the file `GPT-neo-ray-tune.py` in the specified container with the correct arguments.
 
-[comment]: <> (List your hands-on exercises for the lecture here and provide the necessary links to scripts, notebooks, etc. to run them on LUMI)
+4. Specify computational resources and parameter space in `GPT-neo-ray-tune.py`.
 
-* ...
+    * In line 166, pass the correct number of CPUs and GPUs to ray.
```

Comment:
Lukas just mentions to look for `<!!! ACTION REQUIRED ... !!!>` in the code. That may be a bit easier to maintain, since you don't have to remember to check that the line numbers here are correct.